### PR TITLE
INT-B-20465 Customer selecting a counseling office.

### PR DESCRIPTION
**Pull request by JonSpight**
## Summary
When a customer selects an origin duty location assigned to government counseling, then they will also be asked to select a counseling office within the same GBLOC. This field will be called "Counseling Office".

### How to test as a Customer
1.  Login as a Customer
2. Select a counseling Office while creating a new move
3. Check the Moves data table for corresponding counseling_office_id

### How to test as a Service Counselor
1.  Login as a Service Counselor
2. Verify the counseling queue's counseling Office column matches what was created in previous move

## 508 Screenshots
[508Screenshots-20465.pdf](https://github.com/user-attachments/files/16971315/508Screenshots-20465.pdf)

### CODE REVIEW

File: src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx [hunk 1]
```diff
@@ -37,7 +37,21 @@ const OrdersInfoForm = ({ ordersTypeOptions, initialValues, onSubmit, onBack })
     new_duty_location: Yup.object().nullable().required('Required'),
     grade: Yup.mixed().oneOf(Object.keys(ORDERS_PAY_GRADE_OPTIONS)).required('Required'),
     origin_duty_location: Yup.object().nullable().required('Required'),
+    counseling_office_id: Yup.string(),
   });
+  const [office, setOffice] = useState('');
```
Comment (paulstonebraker) [inline src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx:42]: `[dutyLocation, setDutyLocation]` or `[originDutyLocation, setOriginDutyLocation]` or something to that effect

File: src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx [hunk 2]
```diff
@@ -37,7 +37,21 @@ const OrdersInfoForm = ({ ordersTypeOptions, initialValues, onSubmit, onBack })
...
+  const [officeOptions, setOfficeOptions] = useState(null);
+  useEffect(() => {
+    showCounselingOffices(office.id).then((fetchedData) => {
+      if (fetchedData.body) {
+        const transformedItems = fetchedData.body.map((item) => ({
```
Comment (paulstonebraker) [inline src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx:47]: `const counselingOffices` imo, though this won't block me from approving like the above 😸

File: src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx [hunk 3]
```diff
@@ -111,10 +125,21 @@ const OrdersInfoForm = ({ ordersTypeOptions, initialValues, onSubmit, onBack })
                 hint="Required"
                 name="origin_duty_location"
                 id="origin_duty_location"
+                onOfficeChange={(e) => {
```
Comment (paulstonebraker) [inline src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx:128]: `onDutyLocationChange`

File: src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx [hunk 4]
```diff
@@ -111,10 +125,21 @@ const OrdersInfoForm = ({ ordersTypeOptions, initialValues, onSubmit, onBack })
...
+                  setOffice(e);
+                }}
                 required
                 metaOverride={originMeta}
               />
-
+              {office.provides_services_counseling && (
```
Comment (paulstonebraker) [inline src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx:134]: `duty_location.provides_services_counseling`

File: src/components/LocationSearchBox/LocationSearchBox.jsx [hunk 5]
```diff
@@ -113,7 +113,7 @@ export const LocationSearchBoxComponent = ({
   isDisabled,
   handleZipCityOnChange,
 }) => {
-  const { value, onChange, name: inputName } = input;
+  const { value, onChange, officeState, name: inputName } = input;
```
Comment (paulstonebraker) [inline src/components/LocationSearchBox/LocationSearchBox.jsx:116]: `locationState`

File: src/components/Table/SearchResultsTable.jsx [hunk 6]
```diff
@@ -1,26 +1,26 @@
...
+import TextBoxFilter from 'components/Table/Filters/TextBoxFilter';
+import Table from 'components/Table/Table';
 import { CHECK_SPECIAL_ORDERS_TYPES, SPECIAL_ORDERS_TYPES } from 'constants/orders';
+import { BRANCH_OPTIONS, MOVE_STATUS_LABELS, SEARCH_QUEUE_STATUS_FILTER_OPTIONS, SortShape } from 'constants/queues';
+import { servicesCounselingRoutes } from 'constants/routes';
+import { DATE_FORMAT_STRING } from 'shared/constants';
+import LoadingPlaceholder from 'shared/LoadingPlaceholder';
+import SomethingWentWrong from 'shared/SomethingWentWrong';
 import { isBooleanFlagEnabled } from 'utils/featureFlags';
+import { formatDateFromIso, serviceMemberAgencyLabel } from 'utils/formatters';
```
Comment (paulstonebraker) [inline src/components/Table/SearchResultsTable.jsx:23]: Whats with all these changes?

File: src/components/form/fields/DutyLocationInput.jsx [hunk 7]
```diff
@@ -7,10 +7,16 @@ import './DropdownInput.module.scss';
 
 // TODO: refactor component when we can to make it more user friendly with Formik
 export const DutyLocationInput = (props) => {
-  const { label, name, displayAddress, hint, placeholder, isDisabled, searchLocations, metaOverride } = props;
+  const { label, name, displayAddress, hint, placeholder, isDisabled, searchLocations, metaOverride, onOfficeChange } =
+    props;
   const [field, meta, helpers] = useField(props);
 
   const { touched } = useFormikContext();
+  const handleOfficeChange = (event) => {
```
Comment (paulstonebraker) [inline src/components/form/fields/DutyLocationInput.jsx:15]: `handleDutyLocationChange` or something

**Review by paulstonebraker — CHANGES_REQUESTED** [review]
Great stuff! Just did code review - have not tested yet but will do shortly

File: src/components/Table/SearchResultsTable.jsx (hunk 6 above)
Comment (JonSpight) [reply src/components/Table/SearchResultsTable.jsx:23]: I want to say theses came from the merge

File: src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx (hunk 1 above)
Comment (JonSpight) [reply src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx:42]: fixed I chose [dutyLocation, setDutyLocation]

File: src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx (hunk 2 above)
Comment (JonSpight) [reply src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx:47]: Fixed!

File: src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx (hunk 3 above)
Comment (JonSpight) [reply src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx:128]: onOfficeChange has been changed to onDutyLocationChange

File: src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx (hunk 4 above)
Comment (JonSpight) [reply src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx:134]: changed office to dutyLocation to match the above recommended variable name

File: src/components/LocationSearchBox/LocationSearchBox.jsx (hunk 5 above)
Comment (JonSpight) [reply src/components/LocationSearchBox/LocationSearchBox.jsx:116]: changed officeState to locationState

File: src/components/form/fields/DutyLocationInput.jsx (hunk 7 above)
Comment (JonSpight) [reply src/components/form/fields/DutyLocationInput.jsx:15]: updated and fixed

**Pull request closed without merging**